### DISCUSSION TITLE
Is Docker 20.10 EOL?

### DISCUSSION THREAD

**Discussion by yansifw** [moby/moby#45104] [Q&A] [answered]
Based on this https://endoflife.date/docker-engine Docker 20.10 will no receive more security updates? Is that correct?

**Comment by cpuguy83** [comment]
Don't know where that site gets there data, but there is no plan to end support for 20.10 at this time.
I know both Mirantis and Microsoft (my employer) plans to continue support on 20.10.

**Comment by yansifw** [comment]
Looks like it is a "calculated EOL" https://github.com/endoflife-date/endoflife.date/commit/189e5a7eb1cbda36ef143114d76268e89f3e7f43

I also asked in their repo to confirm the source:  https://github.com/endoflife-date/endoflife.date/discussions/2638

**Comment by captn3m0** [comment]
The date was based on the guidance at https://docs.docker.com/engine/install/#support

> Docker Engine releases of a year-month branch are supported with patches as needed for one month after the next year-month general availability release.

We interpret this as 20.10 should be supported one month after 23.0 GA release (which was 2nd February).

I've filed https://github.com/endoflife-date/endoflife.date/pull/2639 to mark 20.10 as a supported branch (Merged, so https://endoflife.date/docker-engine is updated). 

However, how can we get notified once 20.10 (or any similar future branch) goes EOL? Does the support guidance need fixing? Perhaps with a note saying something like "this is general guidance and support might vary for specific branches".

**Comment by Romain-Geissler-1A** [comment]
Hi,

Jumping into this thread, to renew the question from @captn3m0 and to extend it to new branches following the new versioning scheme.

So:
 - Do you know approximately for how many years you plan on keep supporting branch 20.10 ?
 - With the new releasing scheme, which tends to upgrade the major version digit more often than you used to, is there already a predictable support plan that users may refer to ? Typically we know there are new Ubuntu "major" releases every 6 months, supported for a rather "short" time, and every two years there is an LTS ubuntu with X years of guaranteed support.
 
 Cheers,
 Romain

**Reply by YBUSherb** [reply]
> With the new releasing scheme

Where can we learn more about the new release scheme vs the previous one? I was kinda surprised in February with the release of 23.0 when the previous one was 20.10. I also used the "20.10 should be supported +/- 1 month after 23.0 is released" logic and now with 24.0, I'm starting to wonder what is the release scheme and if there is some kind of support lifecycle and LTS that I should focus on

**Reply by cpuguy83** [reply]
I don't have any good answers for this today, but I also don't want to ghost everyone, so...

The current state of things is this:

We have a maintainer from Mirantis (@corhere) and a maintainer from Microsoft (me) who both decided we need to continue to backport bug fixes to 20.10 to support our orgs and we decided to do that in public so as not to duplicate work and so the community as a whole can benefit from the effort.

> for how many years you plan on keep supporting branch 20.10

I don't expect that 20.10 will see much if any more patches at this point, but it depends on what it is.
People should definitely move to at least 23.0 at this point or report issues on why it is be problematic.

re: versioning schemes

We moved to bumping the major version every release primarily because the old format (CalVer) is at odds with the go modules ecosystem.

As an aside, we are just getting the release train rolling again.
The 20.10 release has been a defacto LTS release because we didn't have another release until 2.5 years later.

**Reply by corhere** [reply, marked as answer]
Mirantis will continue to backport security fixes and fixes for bugs which impact Mirantis customers to the 20.10 branch until [Mirantis Container Runtime 20.10 reaches end-of-life](https://docs.mirantis.com/mcr/23.0/compat-matrix/maintenance-lifecycle.html#maintenance-lifecycle) on 2023 December 10.
